server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -275,6 +287,50 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
         Setting.Property.Deprecated
     );
 
+    /**
+     * Sets the cache size for snapshot repository data: the valid range is within 500Kb  ... 1% of the node heap memory.
+     */
+    public static final Setting<ByteSizeValue> SNAPSHOT_REPOSITORY_DATA_CACHE_THRESHOLD = new Setting<>(
+        SNAPSHOT_REPOSITORY_DATA_CACHET_THRESHOLD_SETTING_NAME,
+        CACHE_MAX_THRESHOLD + "b",
+        (s) -> {
+            ByteSizeValue userDefinedLimit = parseBytesSizeValueOrHeapRatio(s, SNAPSHOT_REPOSITORY_DATA_CACHET_THRESHOLD_SETTING_NAME);
+            long userDefinedLimitBytes = userDefinedLimit.getBytes();
+
+            if (userDefinedLimitBytes > CACHE_MAX_THRESHOLD) {
+                throw new OpenSearchParseException(
+                    "{} cannot be larger than [{}] bytes.",
+                    SNAPSHOT_REPOSITORY_DATA_CACHET_THRESHOLD_SETTING_NAME,
+                    CACHE_MAX_THRESHOLD
+                );
+            }
+
+            if (userDefinedLimitBytes < CACHE_MIN_THRESHOLD) {
+                throw new OpenSearchParseException(
```

Comment:
Same, please `IllegalArgumentException`

Reply:
That’s perfect 👍 I’ve made the adjustments!

Reply:
Hello, @reta! I realized I hadn’t pushed this part before going to bed last night 😅
I’ve now pushed it, and updated the default value in `MemorySizeSettingsTests`’s `testSnapshotRepositoryDataCacheSizeSetting` to use `BlobStoreRepository.calculateMaxSnapshotRepositoryDataCacheThreshold()`. Using only `1%` could cause the test to fail in certain environments.